iothub_client/src/uamqp_messaging.c

```diff
@@ -24,6 +24,10 @@
 
 #define MESSAGE_ID_MAX_SIZE 128
 
+#define AMQP_DIAGNOSTIC_ID_KEY "Diagnostic-Id"
+#define AMQP_DIAGNOSTIC_CONTEXT_KEY "Correlation-Context"
+#define AMQP_DIAGNOSTIC_CREATION_TIME_UTC_KEY "creationtimeutc"
```

Comment:
This is the most generic one. That's not a good idea since it will not give much context of what it is for.
Please make sure the name clearly links to E2E diagnostics.

Reply:
Thank you for calling out.
Actually we use the names **Diagnostic-Id** and **Correlation-Context** to align with [trace spec](https://github.com/Microsoft/ApplicationInsights-Home/blob/master/Correlation/Experimental/HttpRequestCorrelationSpec.md) that Microsoft, Google and OpenTracing are trying to setup as industry proposal. As **creationtimeutc**, it is just inner key within **Correlation-Context**.
And these names are already accepted and used in IoT Hub backend team.
Please let me know if you still have concern, thank you!

Reply:
Erich,
when you mention the names are already accepted by the IoT Hub Team, could you elaborate more on that? are they used for other purposes? when were they added?

The spec you linked on your comment is for HTTP protocol. I didn't read any part mentioning such spec should/could spam across other protocols, like AMQP.
Also, for MQTT you used different, specific properties, didn't you?